### DISCUSSION TITLE
Feature Request: ChatGPT-4o

### DISCUSSION THREAD

**Discussion by schnubor** [ueberdosis/tiptap#5146] [Feature Requests]
### Description

Hi!

OpenAI announced their new model yesterday: ChatGPT-4o.

It would be great if it would be added to the supported models of the **AI extension** since it's API pricing is way cheaper compared to ChatGPT-4, making it a rather interesting model.

![CleanShot 2024-05-14 at 09 59 33@2x](https://github.com/ueberdosis/tiptap/assets/5865010/d2837e83-f4d5-4897-b066-e8d76b6662c0)

Cheers,

Christian

### Use Case

AI extension

### Type

New extension

**Comment by nperez0111** [comment]
Hey 👋 !

I think the only thing stopping you from using `gpt-4o` model is the typescript types, if you ignore those it should passthrough and be usable right now. I've opened a PR to fix that!

We are working on a new set of docs so we might not get around to updating these ones.

**Reply by schnubor** [reply]
Nice! Having this typed would obviously be great, but I guess the string to use there would be `gpt-4o`?

**Reply by nperez0111** [reply]
exactly that https://platform.openai.com/docs/models/gpt-4o

**Reply by schnubor** [reply]
Tried it, but it seems the TipTap API doesn't allow this model yet. Zod validation fails:

![CleanShot 2024-05-14 at 17 36 55@2x](https://github.com/ueberdosis/tiptap/assets/5865010/a80a2d09-f6a1-4bfb-9130-4d616d01842d)

![CleanShot 2024-05-14 at 17 37 40@2x](https://github.com/ueberdosis/tiptap/assets/5865010/ccc0e343-1098-427d-8ee7-7dbbc267d93d)

**Reply by nperez0111** [reply]
Ahh I've led you astray, we are working on a version two of the AI backend and I tested against that one. I'll get this fixed!

**Reply by liMingHui-00** [reply]
Why did I encounter ChatGPT is under heavy load when registering chatgpt?

**Reply by nperez0111** [reply]
I'm sorry @liMingHui-00 can you give me more context so I can send your message to the guy who built the ai server?

**Reply by nperez0111** [reply]
And sorry for the delay @schnubor , we are working on a fix. Our deploy failed for some reason 😞

**Reply by liMingHui-00** [reply]
I want to register a gpt account now, but I encountered the problem during the registration process.

**Reply by liMingHui-00** [reply]
The issue is ChatGPT is under heavy load

**Reply by liMingHui-00** [reply]
I can't post pictures here. If I could post pictures, I could better demonstrate the problem.

**Reply by nperez0111** [reply]
We are not OpenAI so unfortunately we would not be of much help for this anyway.

Separately, you can paste images into the github comment editor. Copy the image to your clipboard, and paste into this github comment and it will upload the image for you and send it with the message

**Reply by liMingHui-00** [reply]
thanks

**Comment by schnubor** [comment]
This is now added and working. Closing this :)

Thx @nperez0111 

![CleanShot 2024-05-24 at 09 54 42@2x](https://github.com/ueberdosis/tiptap/assets/5865010/20f13b62-f8c1-465c-bd57-a21f47334f48)

**Reply by nperez0111** [reply]
Ah I forgot to update you, yep!
